Backend/src/main/java/de/neuefische/backend/controllers/ProductController.java

```diff
@@ -0,0 +1,46 @@
+package de.neuefische.backend.controllers;
+
+
+import de.neuefische.backend.entity.Product;
+import de.neuefische.backend.services.ProductService;
+import lombok.AllArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("api/products")
+@AllArgsConstructor
+public class ProductController {
+    private final ProductService productService;
+
+    @GetMapping
+    public List<Product> getAllProducts() {
+        return productService.getAllProducts();
+    }
+
+    @GetMapping("/{id}")
+    public Product getProductById(@PathVariable String id) {
+        return productService.getProductById(id);
+    }
+
+    @GetMapping("/category/{category}")
+    public List<Product> getProductsByCategory(@PathVariable String category) {
+        return productService.getProductsByCategory(category);
+    }
+
+    @PostMapping
+    public void addProduct(@RequestBody Product product) {
```

Comment:
Maybe you should use a DTO (without ID) for the product you receive from the frontend. So the client is not able to set the ID by himself.

Reply:
danke dir Nora